### PR TITLE
DOC/CI: include api docs on travis

### LINKED ISSUE #3800: add travis doc build
not sure if this is worth the trouble but it may catch version incompatibility issues and other bugs not caught by the test suite.

### LINKED ISSUE #6100: DOC/BLD: attributes initialized as None not picked up by autosummary anymore
I already mentioned it here: https://github.com/pydata/pandas/issues/5899#issuecomment-32967854, but then this did not appear on the dev doc builds on the website and only locally on my computer. But now I also see it there. See eg http://pandas.pydata.org/pandas-docs/dev/generated/pandas.DataFrame.html#pandas.DataFrame in the list of attributes, the `is_copy` is no link anymore.

So I get the following warning: `DataFrame.is_copy.rst:: WARNING: document isn't included in any toctree` for 8 methods:
```
pandas.DataFrame.is_copy
pandas.DatetimeIndex.name
pandas.DatetimeIndex.offset
pandas.Index.asi8
pandas.Index.name
pandas.Panel.is_copy
pandas.Panel4D.is_copy
pandas.Series.is_copy
```

Strange thing is, there have not been recent changes to these functions (I think), but the errors only appeared recently. 
@y-p You did the most recent doc build on the website? (does this not happen automatically anymore?)

### PR DESCRIPTION
**Pull request by jorisvandenbossche**
Continued in #12002

---

Closes #6100
Closes #3800

Just to check how this takes on travis

### CODE REVIEW

File: doc/source/api.rst [hunk 1]
```diff
@@ -978,7 +978,7 @@ specific plotting methods of the form ``DataFrame.plot.<kind>``.
 
 .. autosummary::
    :toctree: generated/
-   :template: autosummary/accessor_plot.rst
+   :template: autosummary/accessor_callable.rst
```
Comment (shoyer) [inline doc/source/api.rst]: oops! looks like this snuck through before

**Comment by jreback** [comment]
might simply want to create a doc-build (e.g. this issue https://github.com/pydata/pandas/issues/3800)

**Comment by jorisvandenbossche** [comment]
@jreback yes, I know, but I was first just experimenting a bit to get it working (not very familiar with travis/bash syntax details). Once I have something working the idea was indeed to put it in a separate travis build.

The problem I have now, is that for some reason `make.py` does not output its progress to the standard output of travis (which is the case if you run it locally in a terminal). And then, because of this, travis thinks nothing is happening anymore and aborts the build .. 
Do you know a solution for this?

**Comment by jreback** [comment]
http://docs.travis-ci.com/user/build-timeouts/

might work

**Comment by jreback** [comment]
is make.py writing to stderr? then could just redirect

maybe it's buffering?

**Comment by jreback** [comment]
https://coderwall.com/p/wws2uq/have-travis-ci-test-your-sphinx-docs

?

**Comment by jreback** [comment]
https://github.com/Syntaf/travis-sphinx

**Comment by jorisvandenbossche** [comment]
`make.py` calls `sphinx-build`, which AFAIU outputs to both stdout as stderr? And also, I would think travis outputs both stdout as stderr in the logs? But it is not really clear to me how it is specified the output of the doc build is logged in a file that is then later shown, and why for tests it is outputted directly.

The `travis_wait` is probably not a solution, as this just raises the timeout from 10 to 20 minutes, but I think the doc build will even take longer.

**Comment by jorisvandenbossche** [comment]
aha, hadn't notices that there is also a separate `run_build_docs.sh` script next to `build_docs.sh`, and that is of course capturing that output ..

**Comment by jorisvandenbossche** [comment]
@jreback With the last commit the full docs build OK, although it took 37 minutes. So I put it in a separate travis build, and include the full api docs (there is still a temporary commit to let the others fail).

Can you have a look at the travis ci changes? (it also included somes hacks to numpydoc to prevent some warnings, they are not very nice hacks, but the warnings had to be eliminated to get it running on travis (otherwise the log was too big ..)).

File: .travis.yml [hunk 2]
```diff
@@ -18,6 +18,7 @@ git:
 
 matrix:
     fast_finish: true
+    fail_fast: true
```
Comment (jreback) [inline .travis.yml]: I actually don't think we should have this and leave it false. That way you can see the full gamut of errors (of course for your testsing ok)

File: ci/before_install.sh [hunk 3]
```diff
@@ -11,3 +11,7 @@ echo "inside $0"
 sudo apt-get update $APT_ARGS # run apt-get update for all versions
 
 true # never fail because bad things happened here
+
+if [ x"$DOC_BUILD" == x"" ]; then
+    exit 1
+fi
```
Comment (jreback) [inline ci/before_install.sh]: you nee to put this before the 'true' above. IOW, the script should *always* return true.
(I think this was for testing only?)

File: ci/script.sh [hunk 4]
```diff
@@ -4,6 +4,10 @@ echo "inside $0"
 
 source activate pandas
 
+if [ x"$DOC_BUILD" != x"" ]; then
+    exit 0
```
Comment (jreback) [inline ci/script.sh]: for testing?

File: .travis.yml (hunk 2 above)
Comment (jorisvandenbossche) [inline .travis.yml]: yes, this one is from the temporary commit just for testing, will remove it

File: ci/before_install.sh (hunk 3 above)
Comment (jorisvandenbossche) [inline ci/before_install.sh]: yes, this was only for the testing from the temp commit (just so the others will all fail before installing pandas, so the doc build did not need to wait on all the others)

File: ci/script.sh (hunk 4 above)
Comment (jorisvandenbossche) [inline ci/script.sh]: this one is on purpose and not for testing, so that the doc build does not run the tests (only the docs)

**Comment by jreback** [comment]
@jorisvandenbossche all looks cool to me then. I know the doc-build takes a while but so what its in the  lower section.

note - the builds run in the order that they are defined in the top section (no matter if they are in th ebottom section or not). So you prob want to have the doc-build run last? (I think you have it that way). That way other workers can finish the other jobs. we can have 5 simultaneously i think.

**Comment by jorisvandenbossche** [comment]
The problem with this approach is that the time it takes to build on Travis seems rather variable (maybe depending on its load?). In any case, sometimes it succeeded to finish it in ca 45 min, but other times it goes up to 50min and then travis aborts it. 
But now trying if parallel build speeds it up

**Comment by jreback** [comment]
@jorisvandenbossche closing, but reopen if updating

in theory this looks like a great idea....but yeh it takes quite some time to build the API docs

**Comment by jorisvandenbossche** [comment]
Continued in #12002

**Pull request closed without merging**